### PR TITLE
Add Local Endpoint

### PR DESCRIPTION
**Pull request by SwiftyAlex**
Adds the local endpoint to the app, and connects the local screen to it.

## Changes

Copies the approach as close as I could from the Schedule endpoint, so using nested models etc rather than separate, and using the endpoint name as the top level model name.

To avoid creating `Network` in every model, I've injected it via the Environment so we can have a shared network injected at the top level instead. 

The only visual change is adding an error overlay if the request fails, like this, and a tiny bit of padding so that the content no longer gets obscured by the tab bar.

|🌝|🌚|
| - | - | 
|Simulator Screen Shot - iPhone 13 - 2022-08-08 at 20 22 36|Simulator Screen Shot - iPhone 13 - 2022-08-08 at 20 22 48|

### CODE REVIEW

File: Packages/NetworkKit/Sources/NetworkKit/MockNetwork.swift [hunk 1]
```diff
@@ -32,6 +31,16 @@ public class MockNetwork: Networking {
         .eraseToAnyPublisher()
     }
 
+    public func performRequest<E>(endpoint: E) async throws -> E.DataType where E : Endpoint {
```
Comment (SwiftyAlex) [inline Packages/NetworkKit/Sources/NetworkKit/MockNetwork.swift:34]: I noticed the protocol I copied over was missing the async function from its definition so I added it, and then added the relevant code to MockNetwork

File: SwiftLeeds/Views/Local/BottomSheetView.swift [hunk 2]
```diff
@@ -10,6 +10,9 @@ import SwiftUI
 struct BottomSheetView: View {
     @GestureState private var translation: CGFloat = 0
     @Binding var isOpen: Bool
+
+    let categories: [Local.LocationCategory]
+    let error: Error?
```
Comment (SwiftyAlex) [inline SwiftLeeds/Views/Local/BottomSheetView.swift:15]: I couldn't see an existing approach for error handling so kept it really basic for now as to leave It open for whoever else wants to do it

File: SwiftLeeds/Views/Local/BottomSheetView.swift [hunk 3]
```diff
@@ -34,19 +43,16 @@ struct BottomSheetView: View {
...
+                        ForEach(categories, id: \.self) { category in
+                            LocalCell(
+                                label: category.name,
+                                imageName: category.symbolName,
+                                labelFontStyle: .body
+                            )
+                        }
                     }
+                    .padding(.bottom, Constant.tabBarPadding) // Account for the tab bar
+                    .transition(.opacity)
```
Comment (SwiftyAlex) [inline SwiftLeeds/Views/Local/BottomSheetView.swift:55]: I've added a transition to handle just in-case someone else animates higher up the hierarchy, but I haven't explicitly added an animation in this file

File: SwiftLeeds/Views/Local/BottomSheetView.swift [hunk 4]
```diff
@@ -73,14 +79,26 @@ struct BottomSheetView: View {
     }
 }
 
+private extension BottomSheetView {
+    struct Constant {
+        static let tabBarPadding: CGFloat = UITabBar().frame.height
```
Comment (SwiftyAlex) [inline SwiftLeeds/Views/Local/BottomSheetView.swift:84]: This makes me cringe a bit but I've never found a way around getting the exact number in SwiftUI 🤔

File: SwiftLeeds/Network/ScheduleEndpoint.swift [hunk 5]
```diff
@@ -7,11 +7,6 @@
 
 import NetworkKit
 
-// MARK: - Environment
```
Comment (pdamonkey) [inline SwiftLeeds/Network/ScheduleEndpoint.swift:10]: This should be put back to `Endpoints.swift`, named on purpose to contain all endpoints of the App (as there aren't many and saves lots of small files)

**Review by pdamonkey — CHANGES_REQUESTED** [review]

**Review by pdamonkey — APPROVED** [review]

**Pull request merged**